src/meta/meta_track.c

```diff
@@ -307,12 +307,11 @@ __wt_meta_track_off(WT_SESSION_IMPL *session, bool need_sync, bool unroll)
 		 * should be included in the checkpoint.
 		 */
 		ckpt_session->txn.id = session->txn.id;
-		F_SET(ckpt_session, WT_SESSION_LOCKED_METADATA);
-		WT_WITH_METADATA_LOCK(session,
-		    WT_WITH_DHANDLE(ckpt_session,
-		    WT_SESSION_META_DHANDLE(session),
-		    ret = __wt_checkpoint(ckpt_session, NULL)));
-		F_CLR(ckpt_session, WT_SESSION_LOCKED_METADATA);
+		WT_ASSERT(session,
+		    !F_ISSET(session, WT_SESSION_LOCKED_METADATA));
+		WT_WITH_DHANDLE(ckpt_session, WT_SESSION_META_DHANDLE(session),
+		    WT_WITH_METADATA_LOCK(ckpt_session,
+			ret = __wt_checkpoint(ckpt_session, NULL)));
```

Comment:
I believe this code was structured oddly in case the session passed in already held the metadata lock.  Asserting that isn't the case instead: I can't see the path, but the assertion should fire in testing if I'm wrong.